backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java

```diff
@@ -24,8 +24,13 @@ public void diffAndMerge(AutogenBlob a, AutogenBlob b) throws ModelDBException {
     assertEquals(newB, diffedB);
 
     // Reapplying the diff should not change the result
-    diffedB = DiffMerger.mergeBlob(diffedB, d);
-    assertEquals(newB, diffedB);
+    HashSet<String> conflictSet = new HashSet<String>();
+    diffedB = DiffMerger.mergeBlob(diffedB, d, conflictSet);
+    if (newA == null || newB == null) {
+      assertEquals(newB, diffedB);
+    } else {
+      assertFalse(conflictSet.isEmpty());
```

Comment:
Is there anything we can say about the content of this? Or do you think just testing that it has anything is enough? The goal should be to assert the strongest statement we can with as few assumptions as we can. But maybe building the assert would be too complex for this case?

Reply:
the assert is difficult to generate due to the randomized nature of the inputs.